File: APPR/PW1/src/java/sockets/client/ClientString.java
package java.sockets.client;

import java.io.IOException;
import java.io.PrintStream;
import java.net.Socket;
import java.sockets.Constants;


public class ClientString {
    public static void main(String[] args) throws IOException {
        try (Socket socket = new Socket(Constants.IP_ADDRESS, Constants.PORT)) {
            PrintStream writer = new PrintStream(socket.getOutputStream());
            writer.print("Hello world !");
            writer.flush();
        } finally {
            System.out.println("Message was sent, checkout the server");
        }
    }
}

File: APPR/PW1/src/client/ClientString.kt
package client

import java.io.IOException
import java.io.PrintStream
import java.net.Socket
import java.net.UnknownHostException

fun main() {
    var socket: Socket? = null
    try {
        socket = Socket("localhost", 12345)
        val outStream = socket.getOutputStream()
        val writer = PrintStream(outStream)
        writer.print("Hello world !")
        writer.flush()
    } catch (host: UnknownHostException) {
        println(host.localizedMessage)
    } catch (io: IOException) {
        println(io.localizedMessage)
    } finally {
        socket?.close()
    }
}